scripts/lib/CIME/namelist.py

```diff
@@ -112,7 +112,7 @@
 
 # Fortran syntax regular expressions.
 # Variable names.
-FORTRAN_NAME_REGEX = re.compile(r"(^[a-z][a-z0-9_]{0,62})(\(([+-]?\d*:?){3}\))?$", re.IGNORECASE)
+FORTRAN_NAME_REGEX = re.compile(r"(^[a-z][a-z0-9_]{0,62})(\(((\s*[+-]?\d*\s*:?\s*){3},?\s*){7}\))?$", re.IGNORECASE)
```

Comment:
I am confused by this change. Spaces are _not_ allowed in the identifier to the left of the equal sign (which includes all the array element or array section stuff).